drivers/led/led_gpio.c

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT gpio_leds
+
+/**
+ * @file
+ * @brief GPIO driven LEDs
+ */
+
+#include <drivers/led.h>
+#include <drivers/gpio.h>
+#include <device.h>
+#include <zephyr.h>
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(led_gpio, CONFIG_LED_LOG_LEVEL);
+
+#define DEV_CFG(dev)	((const struct led_gpio_config *) ((dev)->config))
+
+struct led_gpio {
+	const struct device *dev;
+	gpio_pin_t pin;
+	gpio_flags_t flags;
+};
+
+struct led_gpio_config {
+	size_t num_leds;
+	const struct led_gpio *led;
+};
+
+static int led_gpio_set_brightness(const struct device *dev, uint32_t led, uint8_t value)
+{
+
+	const struct led_gpio_config *config = DEV_CFG(dev);
+	const struct led_gpio *led_gpio;
+
+	if ((led >= config->num_leds) || (value > 100)) {
+		return -EINVAL;
+	}
+
+	led_gpio = &config->led[led];
+
+	return gpio_pin_set(led_gpio->dev, led_gpio->pin, (value >= 50));
+}
+
+static int led_gpio_on(const struct device *dev, uint32_t led)
+{
+	return led_gpio_set_brightness(dev, led, 100);
+}
+
+static int led_gpio_off(const struct device *dev, uint32_t led)
+{
+	return led_gpio_set_brightness(dev, led, 0);
+}
+
+static int led_gpio_init(const struct device *dev)
+{
+	const struct led_gpio_config *config = DEV_CFG(dev);
+	int err = 0;
+
+	if (!config->num_leds) {
+		LOG_ERR("%s: no LEDs found (DT child nodes missing)", dev->name);
+		err = -ENODEV;
+	}
+
+	for (size_t i = 0; (i < config->num_leds) && !err; i++) {
+		const struct led_gpio *led = &config->led[i];
+
+		if (device_is_ready(led->dev)) {
+			err = gpio_pin_configure(led->dev, led->pin, led->flags);
```

Comment:
```suggestion
			err = gpio_pin_configure(led->dev, led->pin, GPIO_OUTPUT_INACTIVE | led->flags);
```

Maybe It is worth to add this to be sure that leds will be off after driver initialization

Reply:
ok, added